libbeat/service/service_windows.go

```diff
@@ -71,9 +74,22 @@ func ProcessWindowsControlEvents(stopCallback func()) {
 	if isInteractive {
 		run = debug.Run
 	}
+
 	err = run(os.Args[0], &beatService{})
+
 	if err != nil {
-		logp.Err("Error: %v", err)
+		if errnoErr, ok := err.(syscall.Errno); ok && errnoErr == couldNotConnect {
+			// If, as in the case of Jenkins, the process is started as an interactive process, but the invoking process is itself a service,
+			// beats will incorrectly try to register a service handler. We don't want to swallow errors, so we should still log this, but only as Info.
+			// The only ill effect should be a couple extra idle go routines.
+			//
+			// Ideally we could detect this better, but the only reliable way is with StartServiceCtrlDispatcherW, which is invoked in go with svc.Run. Unfortunately, this also starts some goroutines ahead of time for various reasons. As the docs state for StartServiceCtrlDispatcherW when a 1063 errno is returned:
+			//
+			// "This error is returned if the program is being run as a console application rather than as a service.If the program will be run as a console application for debugging purposes, structure it such that service-specific code is not called when this error is returned."
+			logp.Info("Attempted to register windows service handlers, but this is not a service. No action necessary")
+		} else {
+			logp.Err("Error: %+v", err)
```

Comment:
We should give a little more context in this error message. Maybe something like "Windows service setup failed"?

And it would be good to update this method to using a named logger.

Here's my take on refactoring this method. https://pastebin.com/i9DY6A4K